src/widgets/portainer/component.jsx

```diff
@@ -6,10 +6,57 @@ import useWidgetAPI from "utils/proxy/use-widget-api";
 export default function Component({ service }) {
   const { widget } = service;
 
-  const { data: containersData, error: containersError } = useWidgetAPI(widget, "docker/containers/json", {
-    all: 1,
-  });
+  const isKubernetesWidget = !!widget.kubernetes;
 
+  if (!widget.fields) {
+    widget.fields = isKubernetesWidget ? ["applications", "services", "namespaces"] : ["running", "stopped", "total"];
+  }
+
+  const MAX_ALLOWED_FIELDS = 4;
+  if (widget.fields.length > MAX_ALLOWED_FIELDS) {
+    widget.fields = widget.fields.slice(0, MAX_ALLOWED_FIELDS);
+  }
+
+  const { data: applicationsData, error: applicationsError } = useWidgetAPI(
+    widget,
+    isKubernetesWidget ? "kubernetes/applications" : "",
+  );
+
+  const { data: servicesData, error: servicesError } = useWidgetAPI(
+    widget,
+    isKubernetesWidget ? "kubernetes/services" : "",
+  );
+
+  const { data: namespacesData, error: namespacesError } = useWidgetAPI(
+    widget,
+    isKubernetesWidget ? "kubernetes/namespaces" : "",
+  );
+
+  const { data: containersData, error: containersError } = useWidgetAPI(
+    widget,
+    isKubernetesWidget ? "" : "docker/containers",
+    {
+      all: 1,
+    },
+  );
+
+  // Kubernetes widget handling
+  if (isKubernetesWidget) {
+    const error = applicationsError || servicesError || namespacesError;
```

Comment:
```suggestion
    const error = applicationsError ?? servicesError ?? namespacesError ?? applicationsData;
```

portainer returns error messages in the data